packages/react-components/priority-overflow/src/priorityQueue.ts

```diff
@@ -88,13 +88,14 @@ export function createPriorityQueue<T>(compare: PriorityQueueCompareFn<T>): Prio
   };
 
   const contains = (item: T) => {
-    return arr.indexOf(item) >= 0;
+    const index = arr.indexOf(item);
+    return index >= 0 && index < size;
   };
 
   const remove = (item: T) => {
     const i = arr.indexOf(item);
 
-    if (i === -1) {
+    if (i === -1 || i >= size) {
```

Comment:
How `i` could be bigger than `size`?

Reply:
there is no array 'deletion' happening here, removed items simply get pushed out of bounds